pytorch_lightning/loops/base.py

```diff
@@ -142,9 +240,62 @@ def on_run_end(self) -> Any:
     def teardown(self) -> None:
         """Use to release memory etc."""
 
-    def load_state_dict(self, state_dict: Dict) -> None:
-        """Restore the loop state from the provided state_dict."""
-
     def state_dict(self) -> Dict:
-        """Return the loop current states."""
+        """Current Loop state"""
         return {}
+
+    def load_state_dict(self, state_dict: Dict) -> None:
+        """Reload Loop state"""
+
+    def get_state_dict(self, destination: Optional[OrderedDict] = None, prefix: Optional[str] = '') -> OrderedDict:
```

Comment:
as discussed, could we consider merging several of these state_dict related APIs into one. 
IMO we should aim at keeping the base loop class as simple as possible.